### PR TITLE
Removing creating ModelValidationNode by default in CompositeModelBin…

### PR DESCRIPTION
**Pull request by harshgMSFT**
…der.

This moves the responsibility of of MVN creation to individual model binders if they want the individual models to be validated.

Follow up to this would be cleaning up MVN class to have better (and less error prone constructors).

### CODE REVIEW

**Comment by harshgMSFT** [comment]
Fix line breaks all across.

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs [hunk 1]
```diff
@@ -17,7 +17,10 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
             if (bindingContext.ModelType == typeof(CancellationToken))
             {
                 var model = bindingContext.OperationBindingContext.HttpContext.RequestAborted;
-                return Task.FromResult(new ModelBindingResult(model, bindingContext.ModelName, isModelSet: true));
+                var validationNode =
+                    new ModelValidationNode(bindingContext.ModelName, bindingContext.ModelMetadata, model);
+                return Task.FromResult(
+                    new ModelBindingResult(model, bindingContext.ModelName, isModelSet: true, validationNode: validationNode));
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs]: still long line

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs [hunk 2]
```diff
@@ -37,8 +37,15 @@ public class DefaultObjectValidator : IObjectModelValidator
         /// <inheritdoc />
         public void Validate(
             [NotNull] ModelValidationContext modelValidationContext,
-            [NotNull] ModelValidationNode validationNode)
+            ModelValidationNode validationNode)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs]: Why not just skip the call to this API when there's nothing to validate.

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs (hunk 1 above)
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs]: ill clean up long lines, I somehow missed cleaning up before sending.

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs (hunk 2 above)
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs]: sure

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs [hunk 3]
```diff
@@ -17,7 +17,13 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
             if (bindingContext.ModelType == typeof(CancellationToken))
             {
                 var model = bindingContext.OperationBindingContext.HttpContext.RequestAborted;
-                return Task.FromResult(new ModelBindingResult(model, bindingContext.ModelName, isModelSet: true));
+                var validationNode =
+                    new ModelValidationNode(bindingContext.ModelName, bindingContext.ModelMetadata, model);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs]: Why is there a validation node here? We don't want to validate a `CancellationToken`

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs [hunk 4]
```diff
@@ -46,7 +46,13 @@ public class FormCollectionModelBinder : IModelBinder
                 model = new FormCollection(new Dictionary<string, string[]>());
             }
 
-            return new ModelBindingResult(model, bindingContext.ModelName, isModelSet: true);
+            var validationNode =
+                 new ModelValidationNode(bindingContext.ModelName, bindingContext.ModelMetadata, model);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs]: Same question here, I don't think we want to validate the `IFormCollection`

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs (hunk 3 above)
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs]: Thats y It doesnt has a child node, this is basically for the case when you have a property for cancellation token and there is a [Required] on the property you want to add an error. 
Note this is consistent with what was before.

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs (hunk 4 above)
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs]: same as Cancellation token.

File: src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs (hunk 3 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs]: Discussed in person, makes sense.

File: test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/KeyValuePairModelBinderTest.cs [hunk 5]
```diff
@@ -309,7 +309,9 @@ private static IModelBinder CreateIntBinder()
                 {
                     if (mbc.ModelType == typeof(int))
                     {
-                        return Task.FromResult(new ModelBindingResult(42, mbc.ModelName, true));
+                        var model = 42;
+                        var validationNode = new ModelValidationNode(mbc.ModelName, mbc.ModelMetadata, model);
+                        return Task.FromResult(new ModelBindingResult(model, mbc.ModelName, true, validationNode));
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/KeyValuePairModelBinderTest.cs]: (it's test code but ...) probably worth sticking in the `isModelSet` argument name here and a couple of places below

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs [hunk 6]
```diff
@@ -1462,7 +1462,7 @@ public class ModelBindingTest
 
             };
             var url = "http://localhost/dealers/43/update-vehicle?dealer.name=TestCarDealer&dealer.location=NE";
-            
+
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs]: :100: :dancers: :dango: !!

**Comment by dougbu** [comment]
:shipit:

**Comment by rynowak** [comment]
:shipit:

**Comment by dougbu** [comment]
@harshgMSFT 4f419ee closes this, correct?

**Comment by harshgMSFT** [comment]
Yes. Closed.

**Pull request closed without merging**